From 9aae20500d9cd3e7d55d0536d359bdd1c869db89 Mon Sep 17 00:00:00 2001
From: "Darrick J. Wong" <darrick.wong@oracle.com>
Date: Tue, 30 Oct 2018 10:41:01 +1100
Subject: [PATCH] vfs: skip zero-length dedupe requests
Git-commit: 9aae20500d9cd3e7d55d0536d359bdd1c869db89
Patch-mainline: v4.20-rc1
References: bsc#1133851, bsc#1132219

Don't bother calling the filesystem for a zero-length dedupe request;
we can return zero and exit.

Signed-off-by: Darrick J. Wong <darrick.wong@oracle.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Reviewed-by: Amir Goldstein <amir73il@gmail.com>
Signed-off-by: Dave Chinner <david@fromorbit.com>
Acked-by: Anthony Iliopoulos <ailiopoulos@suse.com>

---
 fs/read_write.c |    5 +++++
 1 file changed, 5 insertions(+)

--- a/fs/read_write.c
+++ b/fs/read_write.c
@@ -2031,6 +2031,11 @@
 	if (!dst_file->f_op->dedupe_file_range)
 		goto out_drop_write;

+	if (len == 0) {
+		ret = 0;
+		goto out_drop_write;
+	}
+
 	ret = dst_file->f_op->dedupe_file_range(src_file, src_pos,
 						len, dst_file, dst_pos);
 out_drop_write:

